pkg/operator/sync.go

```diff
@@ -637,11 +645,105 @@ func (optr *Operator) syncCustomResourceDefinitions() error {
 				return err
 			}
 		}
+
+		if strings.Contains(crd, "controllerconfig") {
+			currentCR, err := optr.apiExtClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "controllerconfigs.machineconfiguration.openshift.io", metav1.GetOptions{})
```

Comment:
suggestion: Hoist everything in the `if strings.Contains(crd, "controllerconfig")` block into a separate function or method so it's a bit easier to follow.

For the section where you `continue`, you can early return nil there to achieve the same effect.